src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -30,7 +30,19 @@ private enum ParameterSyntaxKind : byte
 
         private CustomAttributesBag<CSharpAttributeData> _lazyCustomAttributesBag;
         private ThreeState _lazyHasOptionalAttribute;
-        protected ConstantValue _lazyDefaultSyntaxValue;
+        private ConstantValue _lazyDefaultSyntaxValue;
+
+        private Binder _parameterBinder
+        {
+            get
+            {
+                var isLocalFunction = ContainingSymbol?.Kind == SymbolKind.Method &&
+                    ((MethodSymbol)ContainingSymbol).MethodKind == MethodKind.LocalFunction;
+                return isLocalFunction
+                    ? ((LocalFunctionSymbol)ContainingSymbol).ParameterBinder
+                    : null;
```

Comment:
While we typically prefer `Kind` checks, would it be simpler to use `as` instead here?
```
return (ContainingSymbol as LocalFunctionSymbol)?.ParameterBinder;
```